src/main/java/org/spongepowered/despector/ast/members/insn/arg/NegArg.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.despector.ast.members.insn.arg;
+
+import org.spongepowered.despector.ast.members.insn.InstructionVisitor;
+import org.spongepowered.despector.util.TypeHelper;
+
+/**
+ * An instruction which negates the current value on the stack
+ *
+ * <p>Example: -(val)</p>
+ */
+public class NegArg implements Instruction {
+
+    private final String type;
+    private final Instruction val;
+
+    public NegArg(String type, Instruction val) {
+        this.type = type;
+        this.val = val;
+    }
+
+    public String getType() {
+        return this.type;
+    }
+
+    public Instruction getVal() {
+        return this.val;
+    }
+
+    @Override
+    public String inferType() {
+        return this.type;
```

Comment:
This should use the inferred type of the value that way it stays consistent if the value is modified (I'm in a slow change to make the ast elements mutable rather than immutable as most of them are now).